File: ruby/translator.test.rb
RSpec.describe 'Braille Translator Script' do
  it 'outputs the correct string' do
    output = `ruby translator.rb Abc 123 xYz` # 'ruby translator.rb Abc 234 xYz' should be the corrected command for the given output
    expect(output.strip).to eq('.....OO.....O.O...OO...........O.OOOO.O...OO....OO.O........OO..OO.....OOO.OOOO..OOO')
  end
end

File: javascript/translator.test.js
const { exec } = require('child_process');

describe('translator.js script', () => {
    it('should output correct answer to the console', (done) => {
        exec("node translator.js Abc 123 xYz", (error, stdout, stderr) => {
            expect(error).toBeNull();
            expect(stderr).toBe("");
            expect(stdout.trim()).toBe("O.....O.O...OO...........O......O.O....OO..........OO..OOOO.OOOO..OOO");
            done(); // Call the done callback
        });
    });
});